force-app/main/default/classes/SYS_CustomException.cls

```diff
@@ -0,0 +1,11 @@
+/**
+ * @description       : 
+ * @author            : Jheel
+ * @group             : 
+ * @last modified on  : 04-28-2021
+ * @last modified by  : Jheel
+ * Modifications Log 
+ * Ver   Date         Author   Modification
+ * 1.0   04-28-2021   Jheel   Initial Version
+**/
+public class SYS_CustomException extends Exception {}
```

Comment:
Was this file accidently submitted?

Reply:
No. Actually, I tried to use your Sentry_Exception class itself. But it wasn't providing me with the proper message and Stacktrace. So due to that, I have to create a separate class for the Custom Exception.

Reply:
I didn't see any reference to it?

Also, for the exception, did you instantiate it with the Sentry_ExceptionFactory?

So, the big deal about the Sentry_Exception and Sentry_ExceptionFactory is stack traces.

```
class CustomExceptionClass extends Exception {}
throw new CustomExceptionClass('foo');
```
is great and all, but the stack trace from that is empty.

So to counter that...

```
class CustomExceptionClass extends Sentry_Exception {}
Sentry_Exception ex = (Sentry_Exception) Sentry_ExceptionFactory.build(CustomExceptionClass.class);
ex.setMessage('foo');
```

Also - I don't see any references to it elsewhere in the code, so it looks as though the class itself is unused?

Reply:
Yes. It is not used anywhere in the code. It needs to be used by others to send a custom exception. Even if there is no actual system exception. 
For example, consider a use case in any project, you want to create an issue in Sentry if due to some reason a record is rejected or something. So in that scenario, we aren't running into any exception but we want to create an issue in Sentry so for this we will use this class to throw a custom exception.

So this class is not for our use but more of an outside use class. But if you feel like it's not worth having then you can remove it off. But I would suggest you, to add up some sort of notes in the readme file explaining how to use the Sentry_Exception and Sentry_ExceptionFactory classes.